onebone/signal/filter.py

```diff
@@ -367,3 +367,118 @@ def bandstop_filter(
     b, a = butter(order, [low, high], btype="bandstop")
     signal = lfilter(b, a, signal, axis=axis)
     return signal
+
+
+"""hampel_filter.
+- Author: Sunjin Kim
+- Contact: sunjin.kim@onepredict.com
+"""
+
+
+def hampel_filter(x: np.ndarray, window_size: int, n_sigmas: float = 3) -> Tuple[np.ndarray, list]:
+    """
+    A hampel filter removes outliers.
+    Estimate the median and standard deviation of each sample using
+    MAD(Median Absolute Deviation) in the window range set by the user.
+    If the MAD > 3 * sigma condition is satisfied,
+    the value is replaced with the median value.
+
+    .. math::
+        m_i = median(x_{i-k_{left}}, x_{i-k_{left}+1}, ..., x_{i+k_{right}-1}, x_{i+k_{right}})
+    .. math::
+        MAD_i = median(|x_{i-k}-m_{i}|,...,|x_{i+k}-m_{i}|)
+    .. math::
+        {\sigma}_i = {\kappa} * MAD_i
+
+    Where :math:`k_{left}` and :math:`k_{right}` are the number of neighbors on the left and right sides,
+    respectively, based on x_i (:math:`k_{left} + k_{right}` = window samples).
+    :math:`m_i` is Local median, :math:`MAD_i` is median absolute deviation
+    which is the residuals (deviations) from the data's median.
+    :math:`{\sigma}_i` is the MAD may be used similarly to how one would use the deviation for the average.
+    In order to use the MAD as a consistent estimator
+    for the estimation of the standard deviation :math:`{\sigma}`, one takes :math:`{\kappa} * MAD_i`.
+    :math:`{\kappa}` is a constant scale factor, which depends on the distribution.
+    For normally distributed data :math:`{\kappa}` is taken to be :math:`{\kappa}` = 1.4826
+
+
+    Parameters
+    ----------
+    x : numpy.ndarray
+        1d-timeseries data.
+        The shape of x must be (signal_length,) .
+    window_size : int
+        Lenght of the sliding window.
+        Only integer types are available,
+        and the window size must be adjusted according to your data.
+    n_sigma : float, defalut=3
+        Coefficient of standard deviation.
+
+    Returns
+    ----------
+    filtered_series : numpy.ndarray
```

Comment:
input argument가 `x`이니 output argument도 유사하게 `filtered_x`가 좋을 것 같습니다. 변수 이름 변경 필요.
```suggestion
    filtered_x : numpy.ndarray
```